src/org/labkey/test/tests/SampleTypeTest.java

```diff
@@ -137,6 +138,75 @@ protected void doCleanup(boolean afterTest)
         _userHelper.deleteUsers(false, USER_FOR_FILTERTEST.getEmail());
     }
 
+    // Issue 52390: milliseconds are truncated from time fields on update or reshow
+    @Test
+    public void testDateAndTimeValueUpdates() throws Exception
+    {
+        var projectSettingsPage = goToProjectSettings();
+        projectSettingsPage.setDefaultDateDisplayInherited(false);
+        projectSettingsPage.setDefaultDateDisplay(BaseSettingsPage.DATE_FORMAT.MMMM_dd_yyyy);
+        projectSettingsPage.setDefaultDateTimeDisplayInherited(false);
+        projectSettingsPage.setDefaultDateTimeDisplay(BaseSettingsPage.DATE_FORMAT.dd_MMM_yyyy,
+                BaseSettingsPage.TIME_FORMAT.HH_mm_ss_SSS);
+        projectSettingsPage.setDefaultTimeDisplayInherited(false);
+        projectSettingsPage.setDefaultTimeDisplay(BaseSettingsPage.TIME_FORMAT.HH_mm_ss_SSS);
+        projectSettingsPage.save();
+
+        final String sampleTypeName = "dateTimeEditSamples";
+        final FieldDefinition txtField = new FieldDefinition(
+                TestDataGenerator.randomFieldName("text"), ColumnType.String).setRequired(true);
+        final FieldDefinition dateField = new FieldDefinition(
+                TestDataGenerator.randomFieldName("date", ":"), ColumnType.Date);
+        final FieldDefinition timeField = new FieldDefinition(
+                TestDataGenerator.randomFieldName("time", ":"), ColumnType.Time);
+        final FieldDefinition dateTimeField = new FieldDefinition(
+                TestDataGenerator.randomFieldName("dateTime", ":"),  ColumnType.DateAndTime);
+        final List<FieldDefinition> fields = List.of(txtField, dateField, timeField, dateTimeField);
+
+        SampleTypeDefinition sampleTypeDefinition = new SampleTypeDefinition(sampleTypeName).setFields(fields);
+        sampleTypeDefinition.create(createDefaultConnection(), getProjectName());
+
+        goToProjectHome();
+        waitAndClickAndWait(Locator.linkWithText(sampleTypeName));
+        var dataRegion = DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+        var updatePage = dataRegion.clickInsertNewRow();
+        String date = "January 01 2025";
+        String time = "23:56:54:123";
```

Comment:
Are milliseconds missing below because this is the wrong format?
```suggestion
        String time = "23:56:54.123";
```

Reply:
Yes, it is. Thank you!
(also, I'm a little mad that I missed that)